src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sortedArrayOfSides = [a, b, c].sort((el1, el2) => el2 - el1);
 
-export class Rectangle implements Figure {}
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side length can`t be 0 or less');
+    }
+
+    if (
+      sortedArrayOfSides[0] >=
+      sortedArrayOfSides[1] + sortedArrayOfSides[2]
+    ) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPer = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(
+          semiPer *
+            (semiPer - this.a) *
+            (semiPer - this.b) *
+            (semiPer - this.c),
+        ) * 100,
+      ) / 100
+    );
```

Comment:
[CODE STYLE] - The `getArea` function does not need to accept any parameters. All necessary values are already available as object fields, so there is no need to pass them as parameters.